controllers/rhmiconfig/rhmiconfig_controller.go

```diff
@@ -0,0 +1,174 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	l "github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+
+	croUtil "github.com/integr8ly/cloud-resource-operator/pkg/client"
+	"github.com/integr8ly/integreatly-operator/controllers/rhmiconfig/helpers"
+	k8sErr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	rhmiconfigv1alpha1 "github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+)
+
+var log = l.NewLoggerWithContext(l.Fields{l.ControllerLogContext: "rhmi_config_controller"})
+
+// RHMIConfigReconciler reconciles a RHMIConfig object
+type RHMIConfigReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=integreatly.org,resources=rhmiconfigs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=integreatly.org,resources=rhmiconfigs/status,verbs=get;update;patch
+
+func (r *RHMIConfigReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
```

Comment:
What is the purpose of this?

Reply:
It comes in the Reconcile when scaffolding a new controller. I kept it in case we'd investigate at some point using that context instead of `context.TODO()` as we're currently doing, but it's not being used at this point